### PR TITLE
Task#1 - Implementing simple rest api

### PR DESCRIPTION
**Pull request by warphole**
Good day! I have implemented simple rest api for the first Exadel internship
 homework.  I hope that you will check it and leave your feedback!

### CODE REVIEW

File: src/resources/account/account.router.js [hunk 1]
```diff
@@ -0,0 +1,21 @@
+const { Router } = require('express');
+const {
+  createAccount, deleteAccount, getAccountById, updateAccount,
+} = require('./account.service');
+
+const router = Router();
+// eslint-disable-next-line no-unused-vars
+const httpError = require('../../tools/httpError');
+// eslint-disable-next-line no-empty-pattern
```
Comment (devbarden) [inline src/resources/account/account.router.js:9]: Please configure your eslint configuration. Thats not good to use eslint and ignore the rules

File: src/resources/account/account.router.js [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+router.get('/:id', getAccountById);
+
+router.post('/', createAccount);
+
+// update
```
Comment (devbarden) [inline src/resources/account/account.router.js:15]: no need to add comments in this place. the logic very simple and that is why no need to add some comments here. only if you have some difficult logic then please add some comments

File: src/resources/account/account.router.js [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+router.put('/:id', updateAccount);
+
+// delete
```
Comment (devbarden) [inline src/resources/account/account.router.js:18]: the same

File: src/resources/auth/auth.repository.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
+const bcrypt = require('bcrypt');
+const { users } = require('../../loader/dbconnect');
+// const HttpError = require('../../tools/httpError');
```
Comment (devbarden) [inline src/resources/auth/auth.repository.js:3]: if you don't use this then please remove it

File: src/resources/auth/auth.repository.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+
+function getUserByEmail(email) {
+  return users.find((c) => c.email === email);
+}
+
+function register(user) {
+  users.push({
+    id: Math.random(),
```
Comment (devbarden) [inline src/resources/auth/auth.repository.js:11]: for generate id will be better to use uuid. https://www.npmjs.com/package/uuid

File: src/resources/auth/auth.service.js [hunk 6]
```diff
@@ -0,0 +1,84 @@
...
+  }
+}
+
+async function loginUser(req, res, next) {
+  try {
+    const errors = validationResult(req);
+    if (!errors.isEmpty()) {
+      throw new HttpError('Registration error', 403, errors);
+    }
+    console.log('REQ BODY', req.body);
```
Comment (devbarden) [inline src/resources/auth/auth.service.js:48]: please remove console.log. Its good to use in development but not in production

File: src/resources/auth/auth.service.js [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+      throw new HttpError('Invalid credentials!', 404);
+    }
+  } catch (err) {
+    next(err);
+  }
+}
+
+// eslint-disable-next-line no-unused-vars
+async function logoutUser(req, res) {
+  // some actoin
```
Comment (devbarden) [inline src/resources/auth/auth.service.js:78]: ?

File: src/resources/category/category.router.js [hunk 8]
```diff
@@ -0,0 +1,31 @@
+const { Router } = require('express');
+
+const router = Router();
+// eslint-disable-next-line no-unused-vars
```
Comment (devbarden) [inline src/resources/category/category.router.js:4]: the same about eslint

**Review by devbarden — COMMENTED** [review]
you have empty files, eslint rules ignoring and some style issues. please fix it. otherwise not bad!

File: src/resources/account/account.router.js (hunk 1 above)
Comment (asadbek2021) [reply src/resources/account/account.router.js]: @devbarden  Thank you for your feedback! I have fixed it.

File: src/resources/account/account.router.js (hunk 2 above)
Comment (asadbek2021) [reply src/resources/account/account.router.js]: I got you

File: src/resources/account/account.router.js (hunk 3 above)
Comment (asadbek2021) [reply src/resources/account/account.router.js]: Fixed it

File: src/resources/auth/auth.repository.js (hunk 4 above)
Comment (asadbek2021) [reply src/resources/auth/auth.repository.js]: fixed

File: src/resources/auth/auth.repository.js (hunk 5 above)
Comment (asadbek2021) [reply src/resources/auth/auth.repository.js]: yeap

File: src/resources/auth/auth.service.js (hunk 6 above)
Comment (asadbek2021) [reply src/resources/auth/auth.service.js]: 😄  I forgot remove it

File: src/resources/auth/auth.service.js (hunk 7 above)
Comment (asadbek2021) [reply src/resources/auth/auth.service.js]: removed it

File: src/resources/category/category.router.js (hunk 8 above)
Comment (asadbek2021) [reply src/resources/category/category.router.js]: @devbarden Fixed all of them

**Comment by devbarden** [comment]
if you need you can merge it

**Review by devbarden — APPROVED** [review]

**Pull request merged**